tests/resiliency_tests.rs

```diff
@@ -0,0 +1,314 @@
+            .build()
+            .unwrap();
+
+        let keys = vec!["clean-key1", "clean-key2", "clean-key3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values).collect();
+
+        let mut clean_client = rt.block_on(async {
+            async_memcached::Client::new("tcp://127.0.0.1:11211".to_string())
+                .await
+                .unwrap()
+        });
+
+        for key in &keys {
+            let _ = rt.block_on(async { clean_client.delete(key).await });
+            let result = rt.block_on(async { clean_client.get(key).await });
+            assert_eq!(result, Ok(None));
+        }
+
+        let result = rt.block_on(async { clean_client.set_multi(&kv, None, None).await });
+
+        assert!(result.is_ok());
+    }
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_errors_with_toxic_client_via_with_down() {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let (proxies, toxic_local_addr) = create_proxies_and_configs();
+
+        let toxic_local_url = "tcp://".to_string() + &toxic_local_addr;
+        let toxic_proxy = &proxies[0];
+
+        let keys = vec!["with-down-key1", "with-down-key2", "with-down-key3"];
+        let values = vec!["value1", "value2", "value3"];
+        let kv: Vec<(&str, &str)> = keys.clone().into_iter().zip(values).collect();
+
+        let mut clean_client = rt.block_on(async {
+            async_memcached::Client::new("tcp://127.0.0.1:11211".to_string())
+                .await
+                .unwrap()
+        });
+
+        let mut toxic_client =
+            rt.block_on(async { async_memcached::Client::new(toxic_local_url).await.unwrap() });
+
+        for key in &keys {
+            let _ = rt.block_on(async { clean_client.delete(key).await });
+            let result = rt.block_on(async { clean_client.get(key).await });
+            assert_eq!(result, Ok(None));
+        }
+
+        let _ = toxic_proxy.with_down(|| {
+            rt.block_on(async {
+                let result = toxic_client.set_multi(&kv, None, None).await;
+                assert_eq!(
+                    result,
+                    Err(async_memcached::Error::Io(
+                        std::io::ErrorKind::UnexpectedEof.into()
+                    ))
+                );
+            });
+        });
+    }
+
+    #[ignore = "Relies on a running memcached server and toxiproxy service"]
+    #[test]
+    fn test_set_multi_errors_on_upstream_with_toxic_client_via_limit_data() {
+        let rt = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        let (proxies, toxic_local_addr) = create_proxies_and_configs();
+
+        let toxic_local_url = "tcp://".to_string() + &toxic_local_addr;
```

Comment:
There is a lot of boiler plate code here, can we turn it into some helper methods?

Reply:
Good callout, I'll clean this up 👍

Reply:
Did a refactoring pass on this.  Reconfigured everything so that it's making proper use of environment variables, methods make more sense in the context that they're being use in (e.g. no more unnecessary vecs / `proxies[0]` since we're only ever making one proxy per test) tests should be more parsable and it should be easier to add more tests in the future.